hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/statistics/impl/AwsStatisticsCollector.java

```diff
@@ -69,61 +64,98 @@ public AwsStatisticsCollector(final StatisticsFromAwsSdk collector) {
    * @param response AWS response
    */
   @Override
-  public void collectMetrics(
-      final Request<?> request,
-      final Response<?> response) {
-
-    TimingInfo timingInfo = request.getAWSRequestMetrics().getTimingInfo();
-
-    counter(timingInfo, HttpClientRetryCount.name(),
-        collector::updateAwsRetryCount);
-    counter(timingInfo, RequestCount.name(),
-        collector::updateAwsRequestCount);
-    counter(timingInfo, ThrottleException.name(),
-        collector::updateAwsThrottleExceptionsCount);
-
-    timing(timingInfo, ClientExecuteTime.name(),
-        collector::noteAwsClientExecuteTime);
-    timing(timingInfo, HttpRequestTime.name(),
-        collector::noteAwsRequestTime);
-    timing(timingInfo, RequestMarshallTime.name(),
-        collector::noteRequestMarshallTime);
-    timing(timingInfo, RequestSigningTime.name(),
-        collector::noteRequestSigningTime);
-    timing(timingInfo, ResponseProcessingTime.name(),
-        collector::noteResponseProcessingTime);
+  public void publish(MetricCollection metricCollection) {
+    final long[] throttling = {0};
+    recurseThroughChildren(metricCollection)
+        .collect(Collectors.toList())
+        .forEach(m -> {
+      counter(m, CoreMetric.RETRY_COUNT, retries -> {
+        // Replaces com.amazonaws.util.AWSRequestMetrics.Field.HttpClientRetryCount
+        collector.updateAwsRetryCount(retries);
+
+        // Replaces com.amazonaws.util.AWSRequestMetrics.Field.RequestCount (always HttpClientRetryCount+1)
+        collector.updateAwsRequestCount(retries + 1);
+      });
+
+      // TODO: confirm replacement
+      // Replaces com.amazonaws.util.AWSRequestMetrics.Field.ThrottleException
+      counter(m, HttpMetric.HTTP_STATUS_CODE, statusCode -> {
+        if (statusCode == HttpStatusCode.THROTTLING) {
+          throttling[0] += 1;
+        }
+      });
+
+      // Replaces com.amazonaws.util.AWSRequestMetrics.Field.ClientExecuteTime
```

Comment:
It might be better to remove these comments, and instead create a separate doc explaining these changes, which we can provide when we create the PR with apache. It would also be useful to explain the nested nature of metrics in this doc with an example response as this was not immediately clear to me till I ran in debug mode.